discussions/secure_web_api.md

```diff
@@ -0,0 +1,119 @@
+# L3AFD Secure web API
+
+## Overview
+
+Walmart developed L3AF to simplify the management and orchestration of multiple eBPF programs in an enterprise
+environment.
+
+L3AF’s control plane consists of multiple components that work together to orchestrate eBPF programs:
+
+- L3AF Daemon (L3AFD), which runs on each node where eBPF program runs. L3AFD reads configuration data and manages 
+  the execution and monitoring of eBPF programs running on the node.
+- Deployment APIs, which a user calls to generate configuration data. This configuration data includes which eBPF 
+  programs will run, their execution order, and the configuration arguments for each eBPF program.
+- A database and local key/value (KV) store that stores the configuration data.
+- A datastore that stores eBPF program artifacts (e.g., byte code and/or native code).
+
+When users want to deploy an eBPF program, they can call the L3AFD API with appropriate parameters. This request would
+generate a new config (KV pair). Once L3AFD reads this new config, it orchestrates eBPF programs on the host as per the
+defined parameters. If the user gives a set of eBPF programs, then L3AFD can orchestrate all of them in the sequence
+that the user wanted (aka chaining).
+
+## Need for secure web APIs
+
+L3AFD uses Go HTTPS client to download the configured eBPF packages from a datastore (package repository).
+However, this client does not support yet TLS, but we are investigating how best to support TLS. In the open-source
```

Comment:
This sentence seems to contradict line 24 which says it's an "HTTPS" client.  HTTPS is HTTP over TLS.   If it doesn't support TLS then it's not an HTTPS client, it's only an HTTP client.  Please clarify.

Reply:
This is for eBPF package repo, today we use http client to download.